symcrypt-sys/build/static_link.rs

```diff
@@ -0,0 +1,433 @@
+pbkdf2_hmacsha1.c
+pbkdf2_hmacsha256.c
+pbkdf2.c
+poly1305.c
+primes.c
+rc2.c
+rc4.c
+rdrand.c
+rdseed.c
+recoding.c
+rsa_enc.c
+rsa_padding.c
+rsakey.c
+ScsTable.c
+scsTools.c
+selftest.c
+service_indicator.c
+session.c
+sha1.c
+sha256.c
+sha256Par.c
+sha256Par-ymm.c
+sha256-xmm.c
+sha256-ymm.c
+sha512.c
+sha512Par.c
+sha512Par-ymm.c
+sha512-ymm.c
+sha3.c
+sha3_224.c
+sha3_256.c
+sha3_384.c
+sha3_512.c
+shake.c
+sp800_108_hmacsha1.c
+sp800_108_hmacsha256.c
+sp800_108_hmacsha512.c
+sp800_108.c
+srtp_kdf.c
+srtp_kdf_selftest.c
+ssh_kdf.c
+ssh_kdf_sha256.c
+ssh_kdf_sha512.c
+sskdf.c
+sskdf_selftest.c
+tlsCbcVerify.c
+tlsprf_selftest.c
+tlsprf.c
+xmss.c
+xtsaes.c
+";
+
+fn compile_symcrypt_static(lib_name: &str, options: &SymCryptOptions) -> std::io::Result<()> {
+    // Compile intermediates required this is currently only required for x86_64_unknown_linux_gnu
+    let (already_compiled_files, intermediates) = compile_symcrypt_intermediates(options);
+
+    // Convert already compiled files to a HashSet for faster lookups
+    let already_compiled_set: HashSet<&str> = already_compiled_files.iter().cloned().collect();
+
+    // Prepares list of files to be compiled, excluding already compiled files for x86_64_unknown_linux_gnu
+    let mut base_files: Vec<&'static str> = CMAKE_SOURCES_COMMON
+        .lines()
+        .map(str::trim) // Trim once instead of inside filter
+        .filter(|line| {
+            !line.is_empty() && !line.starts_with("#") && !already_compiled_set.contains(line)
+        })
+        .collect();
+
+    base_files.push("env_generic.c"); // symcrypt_generic
+
+    // Add module-specific files for each target
+    let mut module_files = vec![];
+
+    match options.triple() {
+        Triple::x86_64_pc_windows_msvc | Triple::aarch64_pc_windows_msvc => {
+            base_files.push("env_windowsUserModeWin8_1.c");
+            base_files.push("IEEE802_11SaeCustom.c");
+            module_files.push("inc/static_WindowsDefault.c");
+        }
+        Triple::x86_64_unknown_linux_gnu => {
+            base_files.push("linux/intrinsics.c"); // Only needed for x86_64_unknown_linux_gnu
+            base_files.push("env_posixUserMode.c");
+            module_files.push("inc/static_LinuxDefault.c");
+        }
+        Triple::aarch64_unknown_linux_gnu => {
+            base_files.push("env_posixUserMode.c");
+            module_files.push("inc/static_LinuxDefault.c");
+        }
+    }
+
+    // Add assembly pre generated ASM files to be compiled
+    let asm_files = match options.triple() {
+        Triple::x86_64_pc_windows_msvc => vec![
+            "aesasm.asm",
+            "fdef_asm.asm",
+            "fdef_mulx.asm",
+            "fdef369_asm.asm",
+            "sha256xmm_asm.asm",
+            "sha256ymm_asm.asm",
+            "sha512ymm_asm.asm",
+            "sha512ymm_avx512vl_asm.asm",
+            "wipe.asm",
+        ],
+        Triple::aarch64_pc_windows_msvc => vec!["fdef_asm.asm", "fdef369_asm.asm", "wipe.asm"],
```

Comment:
>["fdef_asm.asm", "fdef369_asm.asm", "wipe.asm"], [](http://example.com/codeflow?start=47&length=48)

I think these are supposed to be *-gas.asm too

Reply:
same as other [comment ](https://github.com/microsoft/rust-symcrypt/pull/81/files/8d5c9a2e9b80daa3d6953ac70a5ae11d40a7a77b#r1956656326)